### ISSUE TITLE
Problem directory name

### ISSUE THREAD

**Issue by sirpatrick2** [doctrine/DoctrineMongoODMModule#225]
The directory DoctrineMongoODMModule/view/laminas-deveoper-tools/toolbar/ has a typo.
laminas-deveoper-tools has a missing 'l'
should be:
laminas-developer-tools

**Comment by rieschl** [comment]
This has been fixed [in the latest PR](https://github.com/doctrine/DoctrineMongoODMModule/pull/222/files#diff-68498889af9f02c8c8402a2ea4575fe28a8a4e231247bac1a2d6bc9b8e9c277f).

**Comment by sirpatrick2** [comment]
Ah ok, sorry I must have done something wrong as composer ~2.0 downloaded version 2.0.2 and I had this typo.

Man thanks.
Patrick


Il 27 ottobre 2020 09:14:42 CET, Thomas Rieschl <notifications@github.com> ha scritto:
>This has been fixed [in the latest
>PR](https://github.com/doctrine/DoctrineMongoODMModule/pull/222/files#diff-68498889af9f02c8c8402a2ea4575fe28a8a4e231247bac1a2d6bc9b8e9c277f).
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/doctrine/DoctrineMongoODMModule/issues/225#issuecomment-717067335

-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by rieschl** [comment]
Well the latest tag is `3.0.0`, so you have to manually update your `composer.json` file and perhaps change something in your code base, as the new version uses now `mongodb-odm` version 2.

**Comment by sirpatrick2** [comment]
OK perfect, thanks

**Comment by sirpatrick2** [comment]
I did install 3.0.0 and changed accordingly my code base.

Still got this error when constructing a form and passing the injected ObjectManager which was retrieved by $container->get('doctrine.documentmanager.odm_default')

 

Argument 1 passed to Arcfot\Form\SchedaEditForm::__construct() must be an instance of Doctrine\Common\Persistence\ObjectManager, instance of Doctrine\ODM\MongoDB\DocumentManager given […]

 

Is that an issue?

 

Many Thanks.

Patrick Invernizzi

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: martedì, 27. ottobre 2020 09:41
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; Manual <manual@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

Well the latest tag is 3.0.0, so you have to manually update your composer.json file and perhaps change something in your code base, as the new version uses now mongodb-odm version 2.

**Comment by rieschl** [comment]
That's not related to this package and neither to `doctrine/mongodb-odm` v2 but `doctrine/persistence`. Apparently, you installed `doctrine/persistence` v2 but didn't migrate your code.

> Argument 1 passed to Arcfot\Form\SchedaEditForm::__construct() must be an instance of Doctrine\Common\Persistence\ObjectManager, instance of Doctrine\ODM\MongoDB\DocumentManager given […]

The namespace has changed to [`Doctrine\Persistence`](https://github.com/doctrine/persistence/blob/3.0.x/UPGRADE-2.0.md#namespace-renamed-to-doctrinepersistence).

TL;DR: change the type hint of your `SchedaEditForm` from `Doctrine\Common\Persistence\ObjectManager` to `\Doctrine\Persistence\ObjectManager`.

**Comment by sirpatrick2** [comment]
I still have persistence v1 but YES, that was exactly the \Common\ wrong!

I really didn’t see it!

 

Thomas, many thanks, you saved my night!

My compliments! Great!

Patrick

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: giovedì, 29. ottobre 2020 20:32
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; State change <state_change@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

That's not related to this package and neither to doctrine/mongodb-odm v2 but doctrine/persistence. Apparently, you installed doctrine/persistence v2 but didn't migrate your code.

Argument 1 passed to Arcfot\Form\SchedaEditForm::__construct() must be an instance of Doctrine\Common\Persistence\ObjectManager, instance of Doctrine\ODM\MongoDB\DocumentManager given […]

The namespace has changed to  <https://github.com/doctrine/persistence/blob/3.0.x/UPGRADE-2.0.md#namespace-renamed-to-doctrinepersistence> Doctrine\Persistence.

TL;DR: change the type hint of your SchedaEditForm from Doctrine\Common\Persistence\ObjectManager to \Doctrine\Persistence\ObjectManager.

**Comment by rieschl** [comment]
> 
> I still have persistence v1 but YES, that was exactly the \Common\ wrong! I really didn’t see it!

Out of curiosity, which version of `doctrine/persistence` are you using, exactly?

**Comment by sirpatrick2** [comment]
I didn’t install it myself, was installed by composer dependencies…

 

In fact I am not sure which version it is, I don’t know how to check it.

Could be that is the 2.1.x so I was wrong.

 

The README in the directory says:

# Doctrine Persistence

 

[![Build Status](https://travis-ci.org/doctrine/persistence.svg)](https://travis-ci.org/doctrine/persistence)

[![Code Coverage](https://codecov.io/gh/doctrine/persistence/branch/2.1.x/graph/badge.svg)](https://codecov.io/gh/doctrine/persistence/branch/2.1.x)

 

The Doctrine Persistence project is a library that provides common abstractions for object mapper persistence.

 

## More resources:

 

* [Website](https://www.doctrine-project.org/)

* [Documentation](https://www.doctrine-project.org/projects/doctrine-persistence/en/latest/index.html)

* [Downloads](https://github.com/doctrine/persistence/releases)

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: giovedì, 29. ottobre 2020 21:21
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; State change <state_change@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

I still have persistence v1 but YES, that was exactly the \Common\ wrong! I really didn’t see it!

Out of curiosity, which version of doctrine/persistence are you using, exactly?

**Comment by rieschl** [comment]
you can check it with `composer info doctrine/persistence`, there you can see the current version

**Comment by sirpatrick2** [comment]
name     : doctrine/persistence 

versions : * 2.1.0

 

sorry, I was wrong…. But hey, really, thanks a lot, I learned something!

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: giovedì, 29. ottobre 2020 21:35
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; State change <state_change@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

you can check it with composer info doctrine/persistence, there you can see the current version

**Comment by sirpatrick2** [comment]
And BTW I managed to make all other changes, so the application runs without errors now!

I was upgrading my app from ZF3 to laminas so I also finally upgraded ODM to 3.0

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: giovedì, 29. ottobre 2020 21:35
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; State change <state_change@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

you can check it with composer info doctrine/persistence, there you can see the current version

**Comment by rieschl** [comment]
> And BTW I managed to make all other changes, so the application runs without errors now! I was upgrading my app from ZF3 to laminas so I also finally upgraded ODM to 3.0

Nice 👍

**Comment by sirpatrick2** [comment]
Thanks a lot to you again, Thomas, and best salutations from Ticino, Southern Switzerland.

Patrick

 

Da: Thomas Rieschl <notifications@github.com> 
Inviato: giovedì, 29. ottobre 2020 22:02
A: doctrine/DoctrineMongoODMModule <DoctrineMongoODMModule@noreply.github.com>
Cc: Patrick <sirpatrick2@gmx.ch>; State change <state_change@noreply.github.com>
Oggetto: Re: [doctrine/DoctrineMongoODMModule] Problem directory name (#225)

 

And BTW I managed to make all other changes, so the application runs without errors now! I was upgrading my app from ZF3 to laminas so I also finally upgraded ODM to 3.0

Nice 👍
